securitycenter/muteconfig/mute_config_test.go

```diff
@@ -0,0 +1,225 @@
+
+package muteconfig
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	securitycenter "cloud.google.com/go/securitycenter/apiv1"
+	"cloud.google.com/go/securitycenter/apiv1/securitycenterpb"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/google/uuid"
+)
+
+var orgID = os.Getenv("SCC_PROJECT_ORG_ID")
+var projectID = os.Getenv("SCC_PROJECT_ID")
+var sourceName = ""
+var finding1Name = ""
+var finding2Name = ""
+
+func createFinding(findingID string, category string, sourceName string) (*securitycenterpb.Finding, error) {
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("securitycenter.NewClient: %v", err)
+	}
+
+	eventTime, err := ptypes.TimestampProto(time.Now())
+	if err != nil {
+		return nil, fmt.Errorf("TimestampProto: %v", err)
+	}
+
+	req := &securitycenterpb.CreateFindingRequest{
+		Parent:    sourceName,
+		FindingId: findingID,
+		Finding: &securitycenterpb.Finding{
+			State: securitycenterpb.Finding_ACTIVE,
+			// Resource the finding is associated with.  This is an
+			// example any resource identifier can be used.
+			ResourceName: fmt.Sprintf("//cloudresourcemanager.googleapis.com/organizations/%s", orgID),
+			// A free-form category.
+			Category: category,
+			// The time associated with discovering the issue.
+			EventTime: eventTime,
+			Severity:  securitycenterpb.Finding_LOW,
+		},
+	}
+	return client.CreateFinding(ctx, req)
+}
+
+func createSource(w io.Writer) error {
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("securitycenter.NewClient: %v", err)
+	}
+	source, err := client.CreateSource(ctx, &securitycenterpb.CreateSourceRequest{
+		Source: &securitycenterpb.Source{
+			DisplayName: "Customized Display Name",
+			Description: "A new custom source that does X",
+		},
+		Parent: fmt.Sprintf("organizations/%s", orgID),
+	})
+
+	if err != nil {
+		return fmt.Errorf("CreateSource: %v", err)
+	}
+	fmt.Fprintf(w, source.Name)
+	return nil
+}
+
+// setupResources initializes variables in this file with resources to
+// use for testing.
+func setupResources() error {
+	orgID = os.Getenv("SCC_PROJECT_ORG_ID")
+	projectID = os.Getenv("SCC_PROJECT_ID")
+	if orgID == "" || projectID == "" {
+		return nil
+	}
+
+	// Create source.
+	buf := &bytes.Buffer{}
+	if err := createSource(buf); err != nil {
+		return fmt.Errorf("createSource: %v", err)
+	}
+	sourceName = buf.String()
+	buf.Reset()
+
+	// Create findings.
+	finding, err := createFinding("updated", "MEDIUM_RISK_ONE", sourceName)
+	if err != nil {
+		return fmt.Errorf("createTestFinding: %v", err)
+	}
+	finding1Name = finding.Name
+	finding, err = createFinding("untouched", "XSS", sourceName)
+	if err != nil {
+		return fmt.Errorf("createTestFinding: %v", err)
+	}
+	finding2Name = finding.Name
+	return nil
+}
+
+func TestMuteConfigCRUD(t *testing.T) {
+	t.Parallel()
```

Comment:
issue: please remove. Unless your tests must run in parallel, this will cause issues.